### ISSUE TITLE
Are the resulting mAP values during training computed over the training set, or the validation set?

### ISSUE THREAD

**Issue by tserdar** [ultralytics/yolov5#4494]
I assumed that it should be on the training set. But then again, running the test.py over training set gives me 15% mAP@.5 higher than it should. Any ideas?

**Comment by glenn-jocher** [comment]
@rocketsfallonrocketfalls https://github.com/ultralytics/yolov5/blob/d1182c4f29e2141be856b85c5d613480a2fffc5d/val.py#L303

**Comment by wudashuo** [comment]
Of course validation set, that's why it calls "validation" set.
If we calculate mAP on the training set, why do we need a validation set?  
If we train on the training set and validate on the training set, the mAP will be 1 or 0.999999

**Comment by tserdar** [comment]
@wudashuo, I have not been sure about the informativeness of active validation mAP feedback during training. 

I mean, mAP does not provide information concerning overfitting for the most part especially if the validation box loss reduces faster than the increase in validation objectness loss in a single class scenario. Therefore, mAP keeps increasing but the model starts to produce false positives.

But I see. Thanks.
